proto/cellar.proto

```diff
@@ -0,0 +1,230 @@
+        // Address of the position to add
+        string position = 2;
+    }
+
+    /*
+    * Push a trusted position to the end of the list of positions used by the cellar. If you
+    know you are going to add a position to the end of the array, this is more efficient then
+    `addPosition`.
+    *
+    * Represents function `pushPosition(address position)`
+    */
+    message PushPosition {
+        // Address of the position to push
+        string position = 1;
+    }
+
+    /*
+    * Remove the position at a given index from the list of positions used by the cellar.
+    *
+    * Represents function `removePosition(uint256 index)`
+    */
+    message RemovePosition {
+        // Index at which to remove the position
+        uint64 index = 1;
+    }
+
+    /*
+    * Set the holding position used by the cellar.
+    *
+    * Represents function `setHoldingPosition(address newHoldingPosition)`
+    */
+    message SetHoldingPosition {
+        // Address of the new holding position to use
+        string new_holding_position = 1;
+    }
+
+    /*
+    * Exchange selector
+    */
+    enum Exchange {
+        EXCHANGE_UNSPECIFIED = 0;
+        // Represents Uniswap V2
+        EXCHANGE_UNIV2 = 1;
+        // Represents Uniswap V3
+        EXCHANGE_UNIV3 = 2;
+    }
+
+    /*
+    * Represents swap parameters for UniswapV2
+    */
+    message UniV2SwapParams {
+        // Array of addresses dictating what swap path to follow
+        repeated string path = 1;
+        // Amount of the first asset in the path to swap
+        string amount = 2;
+        // The minimum amount of the last asset in the path to receive
+        string amount_out_min = 3;
+    }
+
+    /*
+    * Represents swap parameters for UniswapV3
+    */
+    message UniV3SwapParams {
+        // Array of addresses dictating what swap path to follow
+        repeated string path = 1;
+        // Array of pool fees dictating what swap pools to use
+        repeated uint32 pool_fees = 2;
+        // Amount of the first asset in the path to swap
+        string amount = 3;
+        // The minimum amount of the last asset in the path to receive
+        string amount_out_min = 4;
+    }
+
+    /*
+    * Represents swap parameters for an exchange
+    */
+    message SwapParams {
+        oneof params {
+            // Params for a Uniswap V2 swap
+            UniV2SwapParams univ2_params = 1;
+            // Params for a Uniswap V3 swap
+            UniV3SwapParams univ3_params = 2;
+        }
+    }
+
+    /*
+    * Move assets between positions. To move assets from/to this cellar's holdings, specify
+    the address of this cellar as the `fromPosition`/`toPosition`.
+    *
+    * Represents function `rebalance(address fromPosition, address toPosition,
+    *  uint256 assetsFrom, SwapRouter.Exchange exchange, bytes calldata params)
+    */
+    message Rebalance {
+        string from_position = 1;
+        string to_position = 2;
+        string assets_from = 3;
```

Comment:
I suppose here is one example of where we know we need the full `uint256` space and must convert from a string then?

Reply:
Right